spec/MongoStorageAdapter.spec.js

```diff
@@ -350,6 +350,135 @@ describe_only_db('mongo')('MongoStorageAdapter', () => {
     expect(postIndexPlan.executionStats.executionStages.stage).toBe('FETCH');
   });
 
+  it('should never use case_insensitive_email index for regex query on email', async () => {
+    const user = new Parse.User();
+    user.set('username', 'Bugs');
+    user.set('password', 'Bunny');
+    user.set('email', 'bugs@wabbits.com');
+    await user.signUp();
+
+    const database = Config.get(Parse.applicationId).database;
+    const schema = await new Parse.Schema('_User').get();
+
+    await database.adapter.ensureIndex(
+      '_User',
+      schema,
+      ['email'],
+      'case_insensitive_email',
+      true
+    );
```

Comment:
One would think... And that right there is what I spent most of Friday trying to figure out. But if you look at the `ensureIndex` docs, you will also notice that the params it takes and the params being passed here are totally different. So then you might try to use the `createIndex` function anyway and discover that that too takes none of the same props you expected and does nothing you thought it might. After hours of trying to find different docs and thinking you're crazy, you (like me) would discover that `database.adapter.ensureIndex` is not the `MongoDB` package at all! It's an imposter and through the terribly obfuscated file inclusion setup, they hid a file called `MongoStorageAdapter` that has the mysterious `ensureIndex`. Looking into it further, you discover that that function actually implements `createIndex` (on line 721). Your quest is finally over.